packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -46,5 +58,16 @@ export function createProjectService(): TypeScriptProjectService {
     },
     session: undefined,
   });
+
+  return {
+    allowDefaultProjectForFiles: new Set(
+      typeof options === 'object'
+        ? options.allowDefaultProjectForFiles
+            ?.flatMap(pattern => globby.sync(pattern))
+            .map(filePath => path.join(tsconfigRootDir, filePath))
```

Comment:
this should probably use the `getCanonicalFileName` util to ensure it's in the standard form?